users/README.md

```diff
@@ -43,6 +43,7 @@ number and email address.
 |   [GET /users/:userId/badges](#get-usersidbadges)   | Returns badges assigned to the user  |
 |             [POST /users](#post-users)              |          Creates a new User          |
 |        [PATCH /users/self](#patch-usersself)        |       Updates data of the User       |
+|        [GET /users/filter](#get-usersfilter)        |  Returns user data based on a filter |
```

Comment:
This should be `?filter`. (Since `filter` is not a resource)

Reply:
In another issue [here](https://github.com/Real-Dev-Squad/website-backend/issues/966),  `/users/userId`  is being modified to `/users?userId=..` .As this change also accepts query parameters directly after /users.  Adding query parameters here as well after `/users` will cause a collision. 

can we rename this endpoint to `/users/search?...`.

Reply:
@heyrandhir are we going with `/users/filter` or `/users/search` for this API?

Reply:
@vikhyat187  as of now going with /users/search .also waiting for @ankushdharkar comment on this one.

Reply:
Can we just give the results based on the key provided in the query parameters?

If asked for the filter, we filter and give, if asked for ?userId, we give user details.